botocore/client.py

```diff
@@ -534,7 +539,7 @@ def _make_api_call(self, operation_name, api_params):
             http, parsed_response = event_response
         else:
             http, parsed_response = self._endpoint.make_request(
-                operation_model, request_dict)
+                operation_model, request_dict, request_context)
```

Comment:
That context gets discarded when the request is prepared, and this signing context is added in addition to whatever is there. An end user has no way to modify the existing contents of that request context besides changing the signing context we inject.

The request dict thing could work.